WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -161,6 +161,18 @@
       patch :update, id: registration.id, registration: { status: :accepted }
       expect(registration.reload.accepted?).to eq true
     end
+
+    it "can register for their own competition that is not yet visible" do
+      competition.update_column(:showAtAll, false)
+      expect(RegistrationsMailer).to receive(:notify_organizers_of_new_registration).and_call_original
```

Comment:
I think we cannot use `.with(<the_registration>)` in this case, because there's no way to access the registration before it's created below, right?